### PR TITLE
Add Streams API broker compatibility table

### PR DESCRIPTION
**Pull request by JimGalasyn**
Add a broker compatibility matrix to the Streams Developer Guide.

### CODE REVIEW

**Comment by JimGalasyn** [comment]
cc @guozhangwang @mjsax

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -1292,6 +1294,57 @@ <h3 class="anchor-heading"><a id="streams_api_changes_0101" class="anchor-link">
...
+            <th></th>
+            <th colspan="3">Kafka Broker (columns)</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>Kafka Streams API (rows)</td>
+            <td>0.10.0.x</td>
+            <td>0.10.1.x and 0.10.2.x</td>
+            <td>0.11.0.x and<br>1.0.x and<br>1.1.x and<br>2.0.x and<br>2.1.x and<br>2.2.x and<br>2.3.x and<br>2.4.x and<br>2.5.x and<br>2.6.x and<br>2.7.x and<br>2.8.x and<br>3.0.x and<br>3.1.x and<br>3.2.x and<br>3.3.x and<br>3.4.x</td>
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: Seems we need to at 3.5 that was released in the mean time?

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -1292,6 +1294,57 @@ <h3 class="anchor-heading"><a id="streams_api_changes_0101" class="anchor-link">
...
+            <td>compatible</td>
+          </tr>
+          <tr>
+            <td>1.0.x and<br>1.1.x and<br>2.0.x and<br>2.1.x and<br>2.2.0 and<br>2.2.0</td>
+            <td></td>
+            <td>compatible with exactly-once turned off<br>(requires broker version 0.11.0.x or higher);<br>requires message format 0.10 or higher;<br>message headers are not supported<br>(requires broker version 0.11.0.x or higher<br>with message format 0.11 or higher)</td>
+            <td>compatible; requires message format 0.10 or higher;<br>if message headers are used, message format 0.11<br>or higher required</td>
+          </tr>
+          <tr>
+            <td>2.2.1 and<br>2.3.x and<br>2.4.x and<br>2.5.x and<br>2.6.x and<br>2.7.x and<br>2.8.x and<br>3.0.x and<br>3.1.x and<br>3.2.x and<br>3.3.x and<br>3.4.x</td>
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: Same here.

**Review by mjsax — COMMENTED** [review]
Thanks for the PR @JimGalasyn !

**Comment by JimGalasyn** [comment]
Thanks for the review, @mjsax! I added the row/col for `3.5.x`.

**Comment by mjsax** [comment]
Thanks! Merged to `trunk`.

**Pull request merged**